tools/palace-sync.sh

```
@@ -0,0 +1,293 @@
+        fi
+    else
+        echo -e "${YELLOW}⚠️  No sync marker found. Running full scan for first-time setup...${NC}"
+        GIT_LOG_ARGS=""
+        SINCE_LABEL="the beginning (first run)"
+    fi
+fi
+
+echo ""
+
+# --- Collect git log ---
+COMMIT_LOG=$(git -C "$REPO_ROOT" log $GIT_LOG_ARGS \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)